spyder/plugins/variableexplorer/widgets/preferences.py

```diff
@@ -0,0 +1,180 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) Spyder Project Contributors
+#
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+# -----------------------------------------------------------------------------
+
+"""
+Dialog window for setting the viewing preferences of the data frame editor.
+"""
+
+# Standard library import
+from typing import Optional
+
+# Third party imports
+from qtpy.QtWidgets import (
+    QDialog, QDialogButtonBox, QGroupBox, QLabel, QLineEdit, QRadioButton,
+    QVBoxLayout, QWidget)
+
+# Local imports
+from spyder.config.base import _
+
+
+FORMAT_SPEC_URL = (
+    'https://docs.python.org/3/library/string.html'
+    '#format-specification-mini-language'
+)
+
+
+class PreferencesDialog(QDialog):
+    """
+    Dialog window for setting viewing preferences of dataframe or array editor.
+
+    Set the attributes `float_format`, `varying_background` and `global_algo`
+    to set the options, if necessary. Call `exec_()` to show the dialog to the
+    user and allow them to interact. Finally, read the attributes to retrieve
+    the options selected by the user.
+
+    Parameters
+    ----------
+    type_string: str
+        Type of variable being edited; should be 'dataframe' or 'array'.
+        The main difference is that arrays do not support the "by column"
+        coloring algorithm. Some text also uses the type.
+    parent: QWidget, optional
+        Parent widget.
+    """
+
+    def __init__(self, type_string: str, parent: Optional[QWidget] = None):
+        super().__init__(parent)
+        self.type_string = type_string
+
+        self.setWindowTitle(
+            _('{} editor preferences').format(type_string.capitalize())
+        )
+
+        main_layout = QVBoxLayout(self)
+
+        formatting_group = QGroupBox(_('Formatting'))
+        formatting_layout = QVBoxLayout(formatting_group)
+        format_label = QLabel(
+            _('<a href="{url}">Format specification</a> for floats:')
+            .format(url=FORMAT_SPEC_URL)
+        )
+        self.format_input = QLineEdit(self)
+        self.format_input.setToolTip(_(
+            'Use same syntax as for built-in <tt>format()</tt> function. '
+            'Default is <tt>.6g</tt>.'
+        ))
```

Comment:
I think for this and most of the other tooltips you set in this dialog we could the new `TipWidget` in `spyder.widgets.helperwidgets`. That would make finding this additional help easier for users.

What do you think?